nixos/modules/services/networking/sozu.nix

```diff
@@ -0,0 +1,479 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sozu;
+  format = pkgs.formats.toml {};
+
+  globalSettings = format.generate "global.toml" cfg.global;
+  optionalSettings = format.generate "optional.toml" cfg.optional;
+in
+{
+  options.services.sozu = {
+    enable = mkEnableOption "Sozu server";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sozu;
+      description = "Sozu package to use.";
+    };
+
+    configFile = mkOption {
+      type = types.either types.str types.path;
+      default = pkgs.writeText "config.toml" ''
+        ${fileContents globalSettings}
+
+        ${fileContents optionalSettings}
+      '';
+      description = ''
+        Override the configuration file used by Sozu.
+        By default, NixOS generates on automatically.
+      '';
+    };
+
+    global = mkOption {
```

Comment:
You should have a single option called `settings` instead of `global` and `optional`.